include/operations/Operation.hpp

```diff
@@ -154,5 +154,7 @@ class Operation {
 
   virtual void dumpOpenQASM(std::ostream& of, const RegisterNames& qreg,
                             const RegisterNames& creg) const = 0;
+
+  virtual void invert() = 0;
```

Comment:
An alternative would be a `const` function returning a new `Operation` instance. This might be helpful to reduce the code duplication that this PR introduces between `Operations.hpp` and `StandardOperation`.
For the `const` function, `Operations.hpp` could call `invert` without changing the operation itself. Not sure what is better here.

Reply:
I thought a bit about this and I would love to somehow get rid of the code duplication with the `dd/Operations.hpp` file. However, I do not think this comes for free as it would most likely require a copy of the operation in one way or the other. I'd rather leave the code in that file untouched for now and leave a potential refactoring for later.
But it would still be nice to have the respective interfaces for the inverse method ready at that point.

Could we actually offer both methods; an in-place non-const version that does not cause a copy and an const version that essentially calls `op->clone()`, and then `inverse()` on the resulting clone?
It might be a bit tricky to get this working with C++ as C++ does not permit two methods that only differ in their return type last time I checked. But maybe the `const` on one function is enough for the compiler to not complain about it.

Reply:
I think adding a `inverted` function that clones and calls `invert` might work.

Reply:
Sounds worth trying. Maybe it even works when the functions are called the same 🤷🏼

Reply:
I've added an `inverted` function -- for my taste this makes the intent a bit clearer, but I can change it to `invert` if you want!

Reply:
Like that! Might want to call it `getInverse` though in order to be a little more consistent with the rest of the code base and clearly indicate that this is a Getter.